packages/algob/src/internal/deployer.ts

```diff
@@ -470,11 +470,19 @@ export class DeployerDeployMode extends DeployerBasicMode implements Deployer {
   }
 
   async deployASA (
-    name: string,
+    asa: string | { name: string, asaDef: wtypes.ASADef },
```

Comment:
too much overloading is bad. In this case we can easily avoid it:
1. Create two public functions: `deployASA` (current one), `deployASADef (name: string, asaDef: wtypes.ASADef)`
2. `deployASA` can use `deployASADef`.

Reply:
yes, good idea